tests/integration/v3_grpc_test.go

```diff
@@ -1952,3 +1953,77 @@ func waitForRestart(t *testing.T, kvc pb.KVClient) {
 		t.Fatalf("timed out waiting for restart: %v", err)
 	}
 }
+
+func TestV3ReadonlyTxnCancelledContext(t *testing.T) {
```

Comment:
Have you considered to add a common test which is executed on both e2e and integration environments? And also use the new [interface](https://github.com/etcd-io/etcd/blob/main/tests/framework/interface.go) ?  

cc @serathius

Reply:
It seems we can only add integration test here because we need to inject an interceptor, but we do not support that in command line (and also e2e). 

But please consider to use the new interface in https://github.com/etcd-io/etcd/blob/main/tests/framework/integration.go

Reply:
One problem, new interface is not available in v3.5, so using it now would make backport much harder.